atr/docs/user-interface.md

```diff
@@ -115,43 +189,67 @@ The block class also adds a `data-src` attribute to elements, which records whic
 
-The route is decorated with `@route.committer`, which ensures that the route fails before the function is even entered if authentication fails. The function receives a `session` object, which is an instance of [`web.Committer`](/ref/atr/web.py:Committer) with a range of useful properties and methods. The function then loads data, creates a form, checks if the request is a POST, and either processes the form or displays it. After successful processing, it creates a fresh form to clear the data. At the end, it renders a template with all of the variables that the template needs.
+The route is decorated with `@get.committer`, which handles authentication and provides a `session` object that is an instance of [`web.Committer`](/ref/atr/web.py:Committer) with a range of useful properties and methods.
+
+The function builds the UI using an [`htm.Block`](/ref/atr/htm.py:Block) object, which provides a convenient API for incrementally building HTML. The form is rendered directly into the block using [`form.render_block()`](/ref/atr/form.py:render_block), which generates all the necessary HTML with Bootstrap styling.
 
-The template receives the form object and renders it by passing it to one of the `forms.render_*` functions. We previously used Jinja2 macros for this, but are migrating to the new rendering functions in Python (e.g. in [`get/distribution.py`](/ref/atr/get/distribution.py) and [`get/ignores.py`](/ref/atr/get/ignores.py)). The template also receives other data like `asf_id` and `user_committees`, which it uses to display information or make decisions about what to show.
+Finally, the route returns the rendered HTML using [`template.blank()`](/ref/atr/template.py:blank), which renders a minimal template with just a title and content area.
+
+Form submission is handled by a separate POST route:
+```python
+@post.committer("/keys/add")
+@post.form(shared.keys.AddOpenPGPKeyForm)
+async def add(session: web.Committer, add_openpgp_key_form: shared.keys.AddOpenPGPKeyForm) -> web.WerkzeugResponse:
+    """Add a new public signing key to the user's account."""
+    try:
+        key_text = add_openpgp_key_form.public_key
+        selected_committee_names = add_openpgp_key_form.selected_committees
+
+        # Process the validated form data...
+
+        await quart.flash("OpenPGP key added successfully.", "success")
+    except web.FlashError as e:
+        await quart.flash(str(e), "error")
+
+    return await session.redirect(get.keys.keys)
+```
 
-If you use the programmatic rendering functions from [`forms`](/ref/atr/forms.py), you can skip the template entirely. These functions return htpy elements, which you can combine with other htpy elements and return directly from the route, which is often useful for admin routes, for example. You can also use [`template.blank`](/ref/atr/template.py:blank), which renders a minimal template with just a title and content area. This is useful for simple pages that do not need the full template machinery.
```

Comment:
@sbp I took this out because it had to do with the old `forms` module but if we want to keep any info let me know